src/components/BulletCursorOverlay.tsx

```diff
@@ -1,375 +1,157 @@
     if (hideBulletsChildren) return true
     const hideBulletsGrandchildren =
-      value !== '=bullet' && attributeEquals(state, grandchildrenAttributeId, '=bullet', 'None')
+      thought && thought.value !== '=bullet' && attributeEquals(state, grandchildrenAttributeId, '=bullet', 'None')
     if (hideBulletsGrandchildren) return true
     return false
   })
 
-  const children = useSelector<Thought[]>(
-    state => getChildrenRanked(state, head(simplePath)),
-    // only compare id, value, and rank for re-renders
-    equalChildren,
-  )
-
-  const isInContextView = useSelector(state => isContextViewActive(state, parentOf(path)))
-
-  const hideBullet = useHideBullet({
-    children,
-    env: {},
-    hideBulletProp,
-    isEditing: true,
-    simplePath,
-    isInContextView,
-    thoughtId: head(simplePath),
-  })
-
-  return (
-    <PlaceholderTreeNode width={width} x={x} y={y} isTableCol1={isTableCol1}>
-      {showContexts && simplePath?.length > 1 && <PlaceholderContextBreadcrumbs simplePath={simplePath} />}
-
-      <div
-        aria-label='placeholder-thought-container'
-        className={css({
-          /* Use line-height to vertically center the text and bullet. We cannot use padding since it messes up the selection. This needs to be overwritten on multiline elements. See ".child .editable" below. */
-          /* must match value used in Editable useMultiline */
-          lineHeight: '2',
-          // ensure that ThoughtAnnotation is positioned correctly
-          position: 'relative',
-          ...(hideBullet ? { marginLeft: -12 } : null),
-        })}
-      >
-        <CursorOverlay simplePath={simplePath} path={path} leaf={leaf} isInContextView={isInContextView} />
+  // Implement useHideBullet hook - only when we have valid cursor data
+  const shouldHideBullet =
+    cursor && thoughtId && simplePath
+      ? useHideBullet({
+          children: childrenThoughts,
+          env,
+          hideBulletProp: hideBulletAttr,
+          isEditing: true, // keep this as true since cursor overlay only works when editing is true
+          simplePath,
+          isInContextView,
+          thoughtId,
+        })
+      : false
+
+  const prevDomElementRef = React.useRef<HTMLElement | null>(null)
+
+  /** Updates the placeholder node to mirror the current cursor node and target its bullet svg. */
+  const updateCursorNode = useCallback(() => {
+    if (!containerRef.current || !cursor) return
+
+    // Find the cursor TreeNode using the data-cursor attribute
+    const cursorTreeNodes = document.querySelectorAll('div[data-cursor="true"]') as NodeListOf<HTMLElement>
+
+    const cursorTreeNode =
+      cursorTreeNodes.length > 1
+        ? Array.from(cursorTreeNodes).find(node => node !== prevDomElementRef.current)
+        : cursorTreeNodes[0]
+
+    if (!cursorTreeNode) return
+
+    const container = containerRef.current
+
+    // Copy all attributes from the cursor TreeNode to the container (except data-cursor)
```

Comment:
Can you add a comment explaining which attributes are copied and why this is necessary?